documentation/client-credentials.md

```diff
@@ -0,0 +1,103 @@
+# Automating authentication with Client Credentials
+
+One potential issue for scripts and other applications is that it requires user interaction to log in and authenticate.
+The CSS offers an alternative solution for such cases by making use of Client Credentials.
+Once you have created an account as described in the [Identity Provider section](dependency-injection.md),
+it is possible to request a token that can then be used to authenticate without user input.
+
+All requests related to these tokens require you to send along
+the email and password of that account to identify yourself.
+This is a temporary solution until the server has more advanced account management,
+after which this API will change.
+
+Below is example code of how to make use of these tokens.
+It makes use of several utility functions from the 
+[Solid Authentication Client](https://github.com/inrupt/solid-client-authn-js).
+Note that the code below uses `await` at root level which is not possible in JavaScript,
```

Comment:
```suggestion
Note that the code below uses top-level `await`, which not all JavaScript engines support,
```
https://www.stefanjudis.com/today-i-learned/top-level-await-is-available-in-node-js-modules/

Reply:
TIL